### PR TITLE
Add climate_indices

### LINKED ISSUE #243: Can climate oscillation indices regions be added
e.g. Nino 1+2/3/4/3.4, MJO, NAO, etc

### PR DESCRIPTION
**Pull request by ahuang11**
https://github.com/regionmask/regionmask/issues/243

 - [x] Closes #243
 - [ ] Fully documented, including `whats-new.rst`

image

### CODE REVIEW

File: docs/defined_scientific.rst [hunk 1]
```diff
@@ -45,7 +46,17 @@ SREX Regions
...
     @savefig plotting_srex.png width=100%
     plt.tight_layout()
-    
+
+CPC Regions
+============
+
+.. ipython:: python
+
+    regionmask.defined_regions.climate_indices.plot();
```
Comment (mathause) [inline docs/defined_scientific.rst:55]:
```suggestion
    regionmask.defined_regions.climate_indices.plot(label="abbrev");
```
If you switch to one `Regions` instance per index you could combine some in to one plot.

File: docs/defined_scientific.rst [hunk 2]
```diff
@@ -45,7 +46,17 @@ SREX Regions
 
     @savefig plotting_srex.png width=100%
     plt.tight_layout()
-    
+
+CPC Regions
```
Comment (mathause) [inline docs/defined_scientific.rst:50]:
```suggestion
Climate Index regions from the Climate Prediction Center (CPC)
(or similar)

**Review by mathause — COMMENTED** [review]
Thanks for the PR. Unfortunately regionmask cannot handle overlapping regions & the MJO region overlaps with the Niño regions and NINO34 overlaps with NINO3 and NINO4. Also NAO, AO and PNA are the same (is this correct?). Possible solutions

1. Create a `Regions` instance for each of the indices (could potentially combine some of the non overlapping regions)
2. Make regionmask warn if there are overlapping regions and let the user handle it (by subsetting looping over them (#228))
3. Make regionmask handle overlapping regions

I think 1. is the easiest and lead to the fastest turnaround. 2. and 3 need changes to regionmask itself (in another PR) and would thus take longer. Let me know what you think.

File: regionmask/defined_regions/climate_indices.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+outlines[6] = ((180, 20), (-180, 20), (-180, 90), (180, 90))
+outlines[7] = ((180, -20), (-180, -20), (-180, -90), (180, -90))
+outlines[8] = ((180, 20), (-180, 20), (-180, 90), (180, 90))
+outlines[9] = ((180, -15), (-180, -15), (-180, 15), (180, 15))
+
+abbrevs = dict()
+abbrevs[1] = "N34"
+abbrevs[2] = "N03"
+abbrevs[3] = "N04"
+abbrevs[4] = "N12"
```
Comment (mathause) [inline regionmask/defined_regions/climate_indices.py:16]: I think I would abbreviate those as NINO34, NINO3, NINO4 and NINO12. Even if that's not really shorter than the name but I think it's more common. If your abbrevs are commonly used  I am of course fine to leave as is.

**Comment by ahuang11** [comment]
Sorry I was MIA for a while. Can you summarize the next steps for this PR?

**Comment by mathause** [comment]
Good question.

If you are patient you can wait until #228 is implemented. (I don't have a timeline for this but would like to do it as one of the next projects.) If not we can add big warning banners as in #282.

Is it correct that NAO, AO and PNA have the same coords?

**Comment by mathause** [comment]
I started working on #228 (#318) we merge this afterwards. Can you answer the inline comments and confirm that NAO, AO and PNA have the same coords?

I can take care of merging master and finishing this up, unless you want to.

**Comment by mathause** [comment]
Gentle ping @ahuang11

**Comment by ahuang11** [comment]
Sorry I completely missed this For indices, I think NAO is 0-90.

"""
AO index is obtained by projecting the AO loading pattern to the daily anomaly 1000 millibar height field over 20°N-90°N latitude. The AO loading pattern has been chosen as the first mode of EOF analysis using monthly mean 1000 millibar height anomaly data from 1979 to 2000 over 20°N-90°N. For more information on the AO, please see [CPC's AO page](https://www.cpc.ncep.noaa.gov/products/precip/CWlink/daily_ao_index/ao.shtml).
"""


"""
The NAO index is obtained by projecting the NAO loading pattern to the daily anomaly 500 millibar height field over 0-90°N. The NAO loading pattern has been chosen as the first mode of a Rotated Empirical Orthogonal Function (EOF) analysis using monthly mean 500 millibar height anomaly data from 1950 to 2000 over 0-90°N latitude.
"""

"""
The PNA index is obtained by projecting the PNA loading pattern to the daily anomaly 500 millibar height field over 0-90°N. The PNA loading pattern has been chosen as the second mode of a Rotated EOF analysis using monthly mean 500 millibar height anomaly data from 1950 to 2000 over 0-90°N latitude. For more information, please see [CPC's Pacific-North American Pattern](https://www.cpc.ncep.noaa.gov/products/precip/CWlink/ENSO/verf/new.pna.shtml) website.
"""
